### PR TITLE
download functions modified in data.py

### LINKED ISSUE #106: Replace EarthExplorer with SAT API
Replace EarthExplorer with SAT API (https://landsatlook.usgs.gov/sat-api). First we need to check whether we can really find and get the same products there, this is not entirely clear to me on [first sight](https://www.usgs.gov/core-science-systems/nli/landsat/landsat-data-access?qt-science_support_page_related_con=0#qt-science_support_page_related_con).

This goes hand in hand with #103 and #104.

### PR DESCRIPTION
**Pull request by shiwakotisurendra**
closes #106

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ something was changed
 [master] (2021-**-**)
 ---------------------
 - ``data``: fixed download error from EarthExplorer #140
+- ``data``: modified download functions in data module #106
```
Comment (fwfichtner) [inline CHANGELOG.rst:19]: Can you be a bit more descriptive in what you changed?

File: ukis_pysat/data.py [hunk 2]
```diff
@@ -30,6 +35,144 @@
 from ukis_pysat.file import env_get
 from ukis_pysat.members import Datahub
 
+BASE_URL = (
```
Comment (fwfichtner) [inline ukis_pysat/data.py:38]: Can you move this into the `init`?

File: ukis_pysat/data.py [hunk 3]
```diff
@@ -30,6 +35,144 @@
...
+        progress.close()
+
+    if verify:
+        if not compute_md5(fpath) == etag:
+            raise requests.exceptions.HTTPError("Download corrupted.")
+
+    return fpath
+
+
+class Product:
```
Comment (fwfichtner) [inline ukis_pysat/data.py:115]: Please be more descriptive here as well, it is only for Landsat. 
And add you sources!

File: ukis_pysat/data.py [hunk 4]
```diff
@@ -299,10 +442,12 @@ def download_image(self, product_uuid, target_dir):
 
         elif self.src == Datahub.EarthExplorer:
             if not Path(target_dir.joinpath(product_uuid + ".zip")).is_file():
-                from pylandsat import Product
+                # from pylandsat import Product
 
                 # download data from GCS if file does not already exist
                 product = Product(product_uuid)
+                # url = f"{product.baseurl}/{product.product_id}.tif"
+                # download_files(url, target_dir, progressbar=False, verify=True)
```
Comment (fwfichtner) [inline ukis_pysat/data.py:450]: better just delete it instead of commenting out.

**Review by fwfichtner — CHANGES_REQUESTED** [review]
Please make sure the linting works.

If the `UnauthorizedError` stays there we should have a look at it.

**Review by fwfichtner — COMMENTED** [review]
Better, let's keep it open for now though.

**Review by fwfichtner — COMMENTED** [review]
If that works, good.

File: ukis_pysat/data.py [hunk 5]
```diff
@@ -544,7 +544,7 @@ def _prep_aoi(aoi):
                     project = pyproj.Transformer.from_proj(
                         proj_from=pyproj.Proj(aoi.crs["init"]),
                         proj_to=pyproj.Proj("epsg:4326"),
-                        skip_equivalent=True,
+                        skip_equivalent=False,
```
Comment (fwfichtner) [inline ukis_pysat/data.py:547]: Just delete the line please

**Review by fwfichtner — APPROVED** [review]

**Comment by fwfichtner** [comment]
You can also remove `pylandsat` from all requirements here.

**Review by fwfichtner — APPROVED** [review]

**Comment by fwfichtner** [comment]
@MWieland what do you think?
@shiwakotisurendra please add sources according to the license to the code you basically copied.

**Comment by MWieland** [comment]
Looks good to me. Much better then before. Agree with @fwfichtner to add sources of possibly copied code snippets.

File: ukis_pysat/files.json [hunk 6]
```diff
@@ -0,0 +1,113 @@
+{
+ "LC08": [
```
Comment (MWieland) [inline ukis_pysat/files.json:2]: consider the use of enums for platform specific band attributes. connected to #23

Comment (fwfichtner) [reply ukis_pysat/files.json:2]: or [pydantic](https://pydantic-docs.helpmanual.io/)?

Comment (MWieland) [reply ukis_pysat/files.json:2]: you could try something like this (not sure if this works though):
```python
class Platform(Enum):
     Sentinel1 = "Sentinel-1"
     Sentinel2 = "Sentinel-2"

     @property
     def bands(self):
         if self == Platform.Sentinel1:
             return [
                 {"id": 0, "name": "B1", "wavelength": "blue"},
                 {"id": 1, "name": "B2", "wavelength": "green"},
             ]
         elif self == Platform.Sentinel2:
             return [
                 {"id": 0, "name": "B010", "wavelength": "blue"},
                 {"id": 1, "name": "B020", "wavelength": "green"},
             ]

print(Platform.Sentinel1.bands)
```

**Review by fwfichtner — COMMENTED** [review]
@shiwakotisurendra please check the line-endings of tests/test_data.py, I guess you did not change every file.

**Review by fwfichtner — APPROVED** [review]

**Pull request merged**